src/query/storages/fuse/src/operations/analyze.rs

```diff
@@ -49,34 +54,110 @@ impl FuseTable {
+                            mut block_count_sum: u64,
+                            mut row_count_sum: u64,
+                            mut sum_map: HashMap<u32, f64>| {
+                if row_count != 0 {
+                    block_count_sum += 1;
+                    row_count_sum += row_count;
+                    for (i, col_stat) in col_stats.iter() {
+                        let density = match col_stat.distinct_of_values {
+                            Some(ndv) => ndv as f64 / row_count as f64,
+                            None => 0.0,
+                        };
+                        match sum_map.get_mut(i) {
+                            Some(sum) => {
+                                *sum += density;
+                            }
+                            None => {
+                                let _ = sum_map.insert(*i, density);
+                            }
+                        }
+                    }
+                }
+                (block_count_sum, row_count_sum)
+            };
             for segment in segments {
                 let segment = segment?;
-                segment.blocks.iter().for_each(|block| {
-                    let block = block.as_ref();
-                    let row_count = block.row_count;
-                    if row_count != 0 {
-                        block_count_sum += 1;
-                        row_count_sum += row_count;
-                        for (i, col_stat) in block.col_stats.iter() {
-                            let density = match col_stat.distinct_of_values {
-                                Some(ndv) => ndv as f64 / row_count as f64,
-                                None => 0.0,
-                            };
-
-                            match sum_map.get_mut(i) {
-                                Some(sum) => {
-                                    *sum += density;
+                for block in &segment.blocks {
+                    let block = block.as_ref().clone();
+                    let index_location: Option<Location> = block.bloom_filter_index_location;
+                    let index_len = block.bloom_filter_index_size;
+                    if let Some(loc) = index_location {
+                        let row_count = block.row_count;
+                        if row_count != 0 {
+                            let maybe_filter = loc
+                                .read_filter(
+                                    ctx.clone(),
+                                    self.operator.clone(),
+                                    &filter_block_cols,
+                                    index_len,
+                                )
+                                .await;
+                            match maybe_filter {
+                                Ok(filter) => {
+                                    let func_ctx = ctx.try_get_function_context()?;
+                                    if let Some(bloom_index) = BlockFilter::try_create(
```

Comment:
should use `BlockFilter::from_filter_block` here, `BlockFilter::try_create` will recalculate the bloom filter.